app/uk/gov/hmrc/soletraderidentificationfrontend/services/SubmissionService.scala

```diff
@@ -25,61 +25,163 @@ import javax.inject.{Inject, Singleton}
+      else SoleTraderJourneySkippingSaUtrCheck
+    }
+
+  private def calculateSubmissionResponse(journeyId: String,
+                                          businessScenario: BusinessScenario,
+                                          matchingResult: Either[SoleTraderDetailsMatching.SoleTraderDetailsMatchFailure, Boolean],
+                                          journeyConfig: JourneyConfig,
+                                          individualDetails: IndividualDetails)(implicit hc: HeaderCarrier,
+                                                                                ec: ExecutionContext): Future[SubmissionResponse] = businessScenario match {
+    case SoleTraderJourneySkippingSaUtrCheck =>
+      handleSoleTraderJourneySkippingBVCheck(journeyId, matchingResult, journeyConfig.continueUrl, individualDetails.optSautr, individualDetails.optNino)
+    case SoleTraderJourneyWithSaUtrCheck =>
+      handleSoleTraderJourneyWithBVCheck(journeyId, matchingResult, journeyConfig, individualDetails)
+    case IndividualJourney =>
+      handleIndividualJourney(journeyId, matchingResult, journeyConfig.continueUrl)
+  }
+
+  private def handleSoleTraderJourneySkippingBVCheck(journeyId: String,
+                                                     matchingResult: Either[SoleTraderDetailsMatching.SoleTraderDetailsMatchFailure, Boolean],
+                                                     continueUrl: String,
+                                                     maybeSaUtr: Option[String],
+                                                     maybeNino: Option[String]
+                                                    )
+                                                    (implicit hc: HeaderCarrier,
+                                                     ec: ExecutionContext): Future[SubmissionResponse] = matchingResult match {
+    case Right(true) if maybeSaUtr.isDefined =>
+      registrationOrchestrationService
+        .registerWithoutBusinessVerification(journeyId, maybeNino, maybeSaUtr.getOrElse(aSaUtrBug))
+        .map(_ => JourneyCompleted(continueUrl + s"?journeyId=$journeyId"))
+
+    case Right(true | false) if maybeNino.isEmpty && isEnabled(EnableNoNinoJourney) =>
+      for {
+        _ <- createTrnService.createTrn(journeyId)
+        _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+      } yield {
+        JourneyCompleted(continueUrl)
+      }
+
+    case Right(true | false) =>
+      soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+        .map(_ => JourneyCompleted(continueUrl))
+
+    case Left(failure) =>
+      soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+        .map(_ => SoleTraderDetailsMismatch(failure))
+
+  }
+
+  private def aSaUtrBug: Nothing =
+    throw new IllegalStateException("This is a bug. Sole trader journey has to have sautr")
```

Comment:
This isn't a particularly clear message because Sole Traders can have no nino. Maybe Sole trader journey has to have sautr to Register. Or Missing Sautr required to call Register API

Reply:
I agree ... I changed it